ocaml/xapi/message_forwarding.ml

```diff
@@ -2607,6 +2607,32 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
         (fun session_id rpc ->
            Client.Host.get_server_certificate rpc session_id host)
 
+    let install_server_certificate ~__context ~host ~certificate ~private_key ~certificate_chain =
+      let local_fn =
+        Local.Host.install_server_certificate
+          ~host
+          ~certificate
+          ~private_key
+          ~certificate_chain
+      in
+      let success () =
+        (* check that fingerprint of the installed cert matches? *)
+        Some ()
+      in
+
+      let fn () = do_op_on ~local_fn ~__context ~host
+        (fun session_id rpc ->
+          Client.Host.install_server_certificate
```

Comment:
you might want to make this an async call instead

this way the host which is restarting stunnel can let you know whether or not it successfully installed the certificate _before_ it kills all of its stunnel connections

Reply:
I think I'll wait until your PR is in to add async